integration/query_frontend_test.go

```diff
@@ -342,6 +342,79 @@ func runQueryFrontendTest(t *testing.T, cfg queryFrontendTestConfig) {
 	assertServiceMetricsPrefixes(t, QueryScheduler, queryScheduler)
 }
 
+func TestQueryFrontendTooLargeEntryRequest(t *testing.T) {
+	s, err := e2e.NewScenario(networkName)
+	require.NoError(t, err)
+	defer s.Close()
+
+	cfg := &queryFrontendTestConfig{
+		setup: func(t *testing.T, s *e2e.Scenario) (configFile string, flags map[string]string) {
+			flags = mergeFlags(BlocksStorageFlags(), BlocksStorageS3Flags())
+
+			minio := e2edb.NewMinio(9000, flags["-blocks-storage.s3.bucket-name"])
+			require.NoError(t, s.StartAndWaitReady(minio))
+
+			return "", flags
+		},
+	}
+
+	memcached := e2ecache.NewMemcached()
+	consul := e2edb.NewConsul()
+	require.NoError(t, s.StartAndWaitReady(consul, memcached))
+
+	configFile, flags := cfg.setup(t, s)
+
+	flags = mergeFlags(flags, map[string]string{
+		"-query-frontend.cache-results":                     "true",
+		"-query-frontend.results-cache.backend":             "memcached",
+		"-query-frontend.results-cache.memcached.addresses": "dns+" + memcached.NetworkEndpoint(e2ecache.MemcachedPort),
+		"-query-frontend.query-stats-enabled":               strconv.FormatBool(cfg.queryStatsEnabled),
+		"-query-frontend.parallelize-shardable-queries":     "true", // Allow queries to be parallized (query-sharding)
+		// Set the maximum entry size to 50 byte.
+		// The query size is 64 bytes, so it will be a too large entry request.
```

Comment:
I couldn't find big differences between the code in `runQueryFrontendWithQueryShardingAndTooManyRequestsTest` and `TestQueryFrontendTooLargeEntryRequest`. Can you maybe reuse that code? I think we can make the status code configurable in the `queryFrontendTestConfig`.
So the e2e test for this PR will be only adding a subtest in `TestQueryFrontendTooLargeEntryRequest`

I think we should also rename `TestQueryFrontendTooLargeEntryRequest` and `runQueryFrontendWithQueryShardingAndTooManyRequestsTest` in that case to something more generic.

Reply:
Make sense. Updated.

Reply:
I don't enable the queryScheduler for TestQueryFrontendWithQueryShardingAndTooLargeEntryRequest.

Because I found that if you enable query scheduler, the size of `QueryResult` request is bigger than the size of `sum(series_1)`. 

I don't know how to make a bigger query for it(at least bigger than the size of `QueryResult` request). Could you please help me?

Reply:
I see, the response with the error itself is bigger than the query result 😄 

I think running without query scheduler is ok, the error handling is happening between querier and query-frontend, so the query-scheduler shouldn't make a difference.